kernel/rtlil.cc

```diff
@@ -168,25 +170,24 @@ RTLIL::Const RTLIL::Const::from_string(std::string str)
 std::string RTLIL::Const::decode_string() const
 {
 	std::string string;
-	std::vector<char> string_chars;
-	for (int i = 0; i < int (bits.size()); i += 8) {
+	string.reserve(GetSize(bits)/8);
+	for (int i = 0; i < GetSize(bits); i += 8) {
 		char ch = 0;
 		for (int j = 0; j < 8 && i + j < int (bits.size()); j++)
 			if (bits[i + j] == RTLIL::State::S1)
 				ch |= 1 << j;
 		if (ch != 0)
-			string_chars.push_back(ch);
+			string.append({ch});
```

Comment:
Is this `{ch}` in order to use the `std::string::append(const char *)` method?  If so, this is really dangerous and should be `{ch, 0}`.  But really you should probably just use:
```suggestion
			string.append(1, ch);
```

Reply:
No, that's using C++11 an initializer_list.

Reply:
Ah, phew, OK.  Still, isn't it clearer (potentially also to the compiler?) to do `append(1, ch)`, and also more performant?  I don't know enough about C++11 to have confidence, but doesn't the `append({ch})` way require instantiating an `initializer_list<char>` object?